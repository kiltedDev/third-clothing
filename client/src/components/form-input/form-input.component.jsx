import React from 'react';

import './form-input.styles.scss';

export const FormInput = ({ handleChange, label, length, ...otherProps }) => (
  <div className='group'>
    <input
      className='form-input'
      onChange={ handleChange }
      { ...otherProps }
    />
  {
    console.log(otherProps)
  }
    {
      label ?
      (<label className={`${ length ? 'shrink': ''} form-input-label`}>
        { label }
      </label>)
      : null
    }
  </div>
)

export default FormInput;
