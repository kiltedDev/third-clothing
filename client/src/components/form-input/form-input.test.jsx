import { shallow } from 'enzyme';
import React from 'react';
import { FormInput } from './form-input.component';

describe('FormInput', () => {
  let wrapper;
  let snapshot;

  let mockHandleChange;
  beforeAll(() => {
    mockHandleChange = jest.fn();
    const mockProps = {
      handleChange: mockHandleChange
    }
    wrapper = shallow(<FormInput { ...mockProps } />);
    snapshot = wrapper.debug();
  })

  it('should render FormInput component',() => {
    expect( snapshot ).toMatchSnapshot();
  })

  it('should have a lebel when required',() => {
    const mockProps2 = {
      handleChange: mockHandleChange,
      label: 'email'
    }
    const wrapper2 = shallow(<FormInput { ...mockProps2 } />);

    expect( wrapper2.debug() ).toMatchSnapshot();
  })
})
