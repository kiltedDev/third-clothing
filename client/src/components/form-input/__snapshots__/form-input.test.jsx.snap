// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should not have a label when none requested 1`] = `
"<styled.div className=\\"group\\">
  <FormInputContainer className=\\"form-input\\" onChange={[Function: mockConstructor]} />
</styled.div>"
`;

exports[`FormInput should render FormInput component 1`] = `
"<styled.div className=\\"group\\">
  <FormInputContainer className=\\"form-input\\" onChange={[Function: mockConstructor]} />
  <FormInputLabel className=\\" form-input-label\\">
    email
  </FormInputLabel>
</styled.div>"
`;
